cpp/dolfin/fem/FormIntegrals.h

```diff
@@ -53,35 +61,96 @@ class FormIntegrals
   ///    Function to call for tabulate_tensor on a cell
   const std::function<void(PetscScalar*, const PetscScalar*, const double*,
                            int)>&
-  get_tabulate_tensor_fn_cell(int i) const;
+  get_tabulate_tensor_fn_cell(unsigned int i) const;
 
+  /// Get the function for 'tabulate_tensor' for exterior facet integral i
+  /// @param i
+  ///    Integral number
+  /// @returns std::function
+  ///    Function to call for tabulate_tensor on an exterior facet
   const std::function<void(PetscScalar*, const PetscScalar*, const double*, int,
                            int)>&
-  get_tabulate_tensor_fn_exterior_facet(int i) const;
+  get_tabulate_tensor_fn_exterior_facet(unsigned int i) const;
+
+  /// Get the function for 'tabulate_tensor' for interior facet integral i
+  /// @param i
+  ///    Integral number
+  /// @returns std::function
+  ///    Function to call for tabulate_tensor on an interior facet
+  const std::function<void(PetscScalar*, const PetscScalar* w, const double*,
+                           const double*, int, int, int, int)>&
+  get_tabulate_tensor_fn_interior_facet(unsigned int i) const;
+
+  /// Register the function for 'tabulate_tensor' for cell integral i
+  void register_tabulate_tensor_cell(int i, void (*fn)(PetscScalar*,
+                                                       const PetscScalar*,
+                                                       const double*, int));
 
-  /// Set the function for 'tabulate_tensor' for cell integral i
-  void set_tabulate_tensor_cell(int i,
-                                void (*fn)(PetscScalar*, const PetscScalar*,
-                                           const double*, int));
+  /// Register the function for 'tabulate_tensor' for exterior facet integral
+  /// i
+  void register_tabulate_tensor_exterior_facet(
+      int i,
+      void (*fn)(PetscScalar*, const PetscScalar*, const double*, int, int));
 
-  /// Set the function for 'tabulate_tensor' for exterior facet integral
+  /// Register the function for 'tabulate_tensor' for exterior facet integral
   /// i
-  void set_tabulate_tensor_exterior_facet(int i,
-                                          void (*fn)(PetscScalar*,
-                                                     const PetscScalar*,
-                                                     const double*, int, int));
+  void register_tabulate_tensor_interior_facet(
+      int i, void (*fn)(PetscScalar*, const PetscScalar* w, const double*,
+                        const double*, int, int, int, int));
 
   /// Number of integrals of given type
   int num_integrals(FormIntegrals::Type t) const;
 
+  /// Get the IDs of integrals of given type, using -1 for the default integral.
```

Comment:
'using -1 for the default integral.' doesn't make sense is this context as there is no index argument.